### PR TITLE
Add tests for oneapi extension compile-time property_list

### PR DESCRIPTION
**Pull request by kopylovanat**
Added to subfolder extension similarly to how extension was treated in https://github.com/KhronosGroup/SYCL-CTS/pull/200

### CODE REVIEW

File: docker/computecpp/configure.sh [hunk 1]
```diff
@@ -7,4 +7,5 @@ cmake . -G Ninja -B build \
     -DCMAKE_PREFIX_PATH=/sycl \
     -DCMAKE_BUILD_TYPE=Release \
     -DSYCL_CTS_ENABLE_FULL_CONFORMANCE=0 \
+    -DSYCL_ENABLE_EXT_ONEAPI_PROPERTY_LIST_TESTS=0 \
```
Comment (ProGTX) [inline docker/computecpp/configure.sh:10]: Similar here, no need to specify the flags that are (and should be) off by default

**Review by ProGTX — APPROVED** [review]

File: docker/computecpp/configure.sh (hunk 1 above)
Comment (kopylovanat) [reply docker/computecpp/configure.sh]: Removed.

File: tests/extension/oneapi_property_list/property_list_has_property.cpp [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+#elif !defined(SYCL_EXT_ONEAPI_DEVICE_GLOBAL)
+    log.skip("SYCL_EXT_ONEAPI_DEVICE_GLOBAL is not defined, test is skipped");
+#else
+    {
+      using namespace sycl::ext::oneapi;
+      property_list prop_list{device_image_scope_v, implement_in_csr_v<true>,
+                              host_access_v<host_access::access::read>};
+
+      if (!prop_list.has_property<device_image_scope>())
+        FAIL(log, "wrong result for device_image_scope");
```
Comment (tomdeakin) [inline tests/extension/oneapi_property_list/property_list_has_property.cpp:41]: Should we not report something more descriptive like "device_image_scope property should exist"?

Comment (keryell) [reply tests/extension/oneapi_property_list/property_list_has_property.cpp:41]: Yes, we could probably make the messages more explicit.

File: tests/extension/oneapi_property_list/property_list_is_property_list.cpp [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+    log.skip("SYCL_EXT_ONEAPI_PROPERTY_LIST is not defined, test is skipped");
+#elif !defined(SYCL_EXT_ONEAPI_DEVICE_GLOBAL)
+    log.skip("SYCL_EXT_ONEAPI_DEVICE_GLOBAL is not defined, test is skipped");
+#else
+    {
+      using namespace sycl::ext::oneapi;
+      property_list prop_list{device_image_scope_v, implement_in_csr_v<true>};
+
+      if (!std::is_base_of_v<std::true_type,
+                             is_property_list<decltype(prop_list)> >)
```
Comment (keryell) [inline tests/extension/oneapi_property_list/property_list_is_property_list.cpp:42]: Welcome to C++11!
```suggestion
                             is_property_list<decltype(prop_list)>>)
```

File: tests/extension/oneapi_property_list/property_list_is_property_list.cpp [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+        FAIL(log,
+             "is_property_list for property list is not derived from "
+             "std::true_type");
+
+      if (!is_property_list_v<decltype(prop_list)>)
+        FAIL(log, "is_property_list_v for property list is not true");
+
+      if (std::is_base_of_v<std::true_type, is_property_list<A> >)
```
Comment (keryell) [inline tests/extension/oneapi_property_list/property_list_is_property_list.cpp:50]:
```suggestion
      if (std::is_base_of_v<std::true_type, is_property_list<A>>)
```
Can you change all of them in all the files?

**Review by keryell — CHANGES_REQUESTED** [review]

**Review by tomdeakin — APPROVED** [review]

**Review by keryell — APPROVED** [review]
Thanks!

**Pull request merged**